src/components/fields/checkbox-group/checkbox-group.tsx

```diff
@@ -48,6 +51,16 @@ export const CheckboxGroup = (props: IGenericFieldProps<ICheckboxGroupSchema>) =
 		// eslint-disable-next-line react-hooks/exhaustive-deps
 	}, [validation]);
 
+	useDeepCompareEffect(() => {
+		const updatedValues = [];
+		options.forEach((option) => {
+			if (value?.includes(option.value)) {
+				updatedValues.push(option.value);
+			}
+		});
+		setValue(id, updatedValues);
```

Comment:
using `setValue` rather than `onChange` to prevent triggering validation
for more complex use cases, devs should use the `setValue()` method exposed through `FrontendEngine` instead